automation/TaskManager.py

```diff
@@ -243,20 +243,20 @@ def _manager_watchdog(self) -> None:
             # 300 second buffer to avoid killing freshly launched browsers
             # TODO This buffer should correspond to the maximum spawn timeout
             if self.process_watchdog:
-                browser_pids: Set[int] = set()
+                geckodriver_pids: Set[int] = set()
                 display_pids: Set[int] = set()
                 check_time = time.time()
                 for browser in self.browsers:
-                    if browser.browser_pid is not None:
-                        browser_pids.add(browser.browser_pid)
+                    if browser.geckodriver_pid is not None:
+                        geckodriver_pids.add(browser.geckodriver_pid)
                     if browser.display_pid is not None:
                         display_pids.add(browser.display_pid)
                 for process in psutil.process_iter():
                     if process.create_time() + 300 < check_time and (
                             (process.name() == 'firefox' and (
-                                process.pid not in browser_pids)) or (
+                                process.pid not in geckodriver_pids)) or (
                             process.name() == 'Xvfb' and (
-                                process.pid not in browser_pids))):
+                                process.pid not in geckodriver_pids))):
```

Comment:
```suggestion
                                process.pid not in display_pids))):
```
Isn't this the thing we want to check here @birdsarah

Reply:
Not @birdsarah, but yes. We want to use display_pids here.